src/huggingface_hub/templates/datasetcard_template.md

```diff
@@ -0,0 +1,126 @@
+---
+{{ card_data }}
+---
+
+# Dataset Card for [Dataset Name]
+
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Dataset Description](#dataset-description)
+  - [Dataset Summary](#dataset-summary)
+  - [Supported Tasks and Leaderboards](#supported-tasks-and-leaderboards)
+  - [Languages](#languages)
+- [Dataset Structure](#dataset-structure)
+  - [Data Instances](#data-instances)
+  - [Data Fields](#data-fields)
+  - [Data Splits](#data-splits)
+- [Dataset Creation](#dataset-creation)
+  - [Curation Rationale](#curation-rationale)
+  - [Source Data](#source-data)
+  - [Annotations](#annotations)
+  - [Personal and Sensitive Information](#personal-and-sensitive-information)
+- [Considerations for Using the Data](#considerations-for-using-the-data)
+  - [Social Impact of Dataset](#social-impact-of-dataset)
+  - [Discussion of Biases](#discussion-of-biases)
+  - [Other Known Limitations](#other-known-limitations)
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+  - [Contributions](#contributions)
```

Comment:
I would remove this (same for model card) given that we display a TOC in the UX (for both datasets and models now) so I don't think those are needed (cc @osanseviero)

## dataset card UI:

<img width="451" alt="image" src="https://user-images.githubusercontent.com/326577/180997103-88c0d753-665b-487f-bb8b-cae39f009f98.png">

## model card UI:

<img width="546" alt="image" src="https://user-images.githubusercontent.com/326577/180997179-181cf4a4-409f-4c3d-89ac-3d7e20eb51a9.png">

Reply:
(also cc @gary149 )

Reply:
(can be in a more relevant PR, of course)

Reply:
Yes, there was a huge discussion around this in #model-cards that they want the ToC both in UX and in the model card content. The logic behind it was that many institutions print model cards and having them in the content is important

Reply:
any way you have a link to the convo handy? 

Usually what happens on website is that you can specify that the table of contents gets printed with some CSS rules like `@media print { ... }`

ie. the webpage looks different when printed vs. on the Web

WDYT?

Reply:
That works for me!

Reply:
(this is unrelated to this PR so feel free to go ahead and merge)